st2actions/st2actions/resultstracker.py

```diff
@@ -0,0 +1,147 @@
+        # task and the work does not get queued behind any single large item.
+        consumer.qos(prefetch_count=1)
+        return [consumer]
+
+    def process_task(self, body, message):
+        # LOG.debug('process_task')
+        # LOG.debug('     body: %s', body)
+        # LOG.debug('     message.properties: %s', message.properties)
+        # LOG.debug('     message.delivery_info: %s', message.delivery_info)
+        try:
+            self._dispatcher.dispatch(self._do_process_task, body)
+        finally:
+            message.ack()
+
+    def _do_process_task(self, body):
+        try:
+            self._add_to_querier(body)
+        except:
+            LOG.exception('Add query_context failed. Message body : %s', body)
+
+    def _add_to_querier(self, body):
+        querier = self._tracker.get_querier(body.query_module)
+        context = QueryContext.from_model(body)
+        querier.add_queries(query_contexts=[context])
+        return
+
+
+class ResultsTracker(object):
+    def __init__(self, q_connection=None):
+        self._queue_consumer = ActionStateQueueConsumer(q_connection, self)
+        self._consumer_thread = None
+        self._queriers = {}
+        self._query_threads = []
+        self._failed_imports = set()
+
+    def start(self):
+        self._bootstrap()
+        self._consumer_thread = eventlet.spawn(self._queue_consumer.run)
+        self._consumer_thread.wait()
+        for thread in self._query_threads():
+            thread.wait()
+
+    def shutdown(self):
+        LOG.info('Tracker shutting down. Stats from queriers:')
+        self._print_stats()
+        self._queue_consumer.shutdown()
+
+    def _print_stats(self):
+        for name, querier in six.iteritems(self._queriers):
+            if querier:
+                querier.print_stats()
+
+    def _bootstrap(self):
+        all_states = ActionExecutionState.get_all()
+        LOG.info('Found %d pending states in db.' % len(all_states))
+
+        query_contexts_dict = defaultdict(list)
+        for state_db in all_states:
+            try:
+                context = QueryContext.from_model(state_db)
+            except:
+                LOG.exception('Invalid state object: %s', state_db)
+                continue
+            query_module_name = state_db.query_module
+            querier = self.get_querier(query_module_name)
+
+            if querier is not None:
+                query_contexts_dict[querier].append(context)
+
+        for querier, contexts in six.iteritems(query_contexts_dict):
+            LOG.info('Found %d pending actions for query module %s', len(contexts), querier)
+            querier.add_queries(query_contexts=contexts)
+
+    def get_querier(self, query_module_name):
+        if (query_module_name not in self._queriers and
+                query_module_name not in self._failed_imports):
+            try:
+                querier = self._import_query_module(query_module_name)
```

Comment:
suggestion -

querier -> query_module
instance -> querier

self._queriers contains the instance therefore the variable names will end up aligning.